### ISSUE TITLE
Support folding function expressions

### ISSUE THREAD

**Issue by Fernandomr88** [Dart-Code/Dart-Code#4723]
@DanTup should I open an issue for this case too? or is it meant to be like this?
There's no folding on these cases:

![image](https://github.com/Dart-Code/Dart-Code/assets/14304128/f1ede756-03b9-4d8f-8ff8-a404a157e50b)

_Originally posted by @Fernandomr88 in https://github.com/Dart-Code/Dart-Code/issues/4506#issuecomment-1706978579_

As I commented on the above, there are some cases where some Flutter Widget parameters are large functions that would be nice to be able to fold.

My proposal, within the limits of VSCode as stated by @DanTup on the above mentioned issue, is to implement the ability to fold those cases too.

**Comment by DanTup** [comment]
Actually, that folding was for the function arguments, not the named parameter:

![image](https://github.com/Dart-Code/Dart-Code/assets/1078012/1c8b078e-ee96-4c81-9c4e-b58d0ef15355)

**Comment by Fernandomr88** [comment]
That will be a game changer for Flutter (since widgets have multiple params that can have many many lines)

**Comment by DanTup** [comment]
Working on a fix at https://dart-review.googlesource.com/c/sdk/+/350581

Example:

https://github.com/Dart-Code/Dart-Code/assets/1078012/64592879-f0ce-4281-867f-b8d157d0d414

**Comment by DanTup** [comment]
Fixed by https://github.com/dart-lang/sdk/commit/5e1fb3ebb915bbd6a0c13183bcaf41294bde555b. The change is in the Dart SDK so will show up in a future SDK release (note a Dart-Code release).

**Comment by Fernandomr88** [comment]
@DanTup why would this case be unfoldable? Is it the expected behavior? 

The `onChanged` below.
<img width="318" alt="image" src="https://github.com/Dart-Code/Dart-Code/assets/14304128/69912978-a7d8-4e70-a9fd-0cc0d21510f0">

**Comment by DanTup** [comment]
@Fernandomr88 if you're on Flutter stable, the fix above has not yet shipped in an SDK release. If you're on Flutter master, could you file an issue with a copy/pastable example I can test with? Thanks!

**Comment by Fernandomr88** [comment]
> @Fernandomr88 if you're on Flutter stable, the fix above has not yet shipped in an SDK release. If you're on Flutter master, could you file an issue with a copy/pastable example I can test with? Thanks!

I'm on stable. Ok, any ETA on that?

**Comment by DanTup** [comment]
Good to know, thanks!

**Comment by Fernandomr88** [comment]
@DanTup is this on stable already? A few dart/flutter releases came since.

**Comment by DanTup** [comment]
@Fernandomr88 yep, this shipped in a stable release 3.22 in May. I just tested an example like your original code here and it seemed to work fine:

https://github.com/user-attachments/assets/3aaab392-67a1-4e79-b1ab-1be71f48f9cc

I tested with code like this:

```dart
Center(
          child: TextButton(
            onPressed: () {
              print('click!');
            },
            child: Text('test'),
          ),
        )
```

If that's not working for you on 3.22, let me know. If it's a different case where you're not seeing folding, please open a new issue with a code sample. Thanks!
